plaso/containers/artifacts.py

```diff
@@ -66,6 +67,120 @@ def __init__(self, name=None, schema='DNS'):
     self.schema = schema
 
 
+class OperatingSystemArtifact(ArtifactAttributeContainer):
+  """Operating system artifact attribute container.
+
+  Attributes:
+    name (str): name, such as "Linux", "MacOS" or "Windows".
+    product (str): product name, such as "macOS Mojave" or "Windows XP".
+    version (str): version, such as "10.14.1" or "5.1".
+  """
+  CONTAINER_TYPE = 'operating_system'
+
+  # Note that the product names below are normalized.
+  _NAMES_AND_VERSIONS_PER_PRODUCT = {
+      'Windows 2000': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 0)),
+      'Windows 2003': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 2)),
+      'Windows 2003 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 2)),
+      'Windows 2008': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 0)),
+      'Windows 2008 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 1)),
+      'Windows 2012': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 2)),
+      'Windows 2012 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 3)),
+      'Windows 2016': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows 2019': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows 7': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 1)),
+      'Windows 8': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 2)),
+      'Windows 8.1': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 3)),
+      'Windows 10': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows Vista': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 0)),
+      'Windows XP': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 1))}
+
+  def __init__(self, name=None, product=None, version=None):
+    """Initializes an operating system artifact.
+
+    Args:
+      name (str): name, such as "Linux", "MacOS" or "Windows".
+      product (str): product name, such as "macOS Mojave" or "Windows XP".
+      version (str): version, such as "10.14.1" or "5.1".
+    """
+    super(OperatingSystemArtifact, self).__init__()
+    self.name = name
+    self.product = product
+    self.version = version
+
+  @property
+  def normalized_product(self):
+    """str: normalized product name."""
```

Comment:
Is what a "normalized product name" is defined somewhere? Without reading the method logic here, I have no idea what this value would be. Please add either a description of what this is here, or mention where one might find a definition.

Reply:
It's a normalized version of the product name, it is unclear to me how a formal definition is going to help?

Instead could you explain what is not clear about the logic below, maybe that could be changed instead?